cli/src/commands/parse.rs

```diff
@@ -113,13 +138,8 @@ pub fn parse_lockfile(
     if !generate_lockfiles || !parser.is_path_manifest(&path) {
         // Return the original lockfile parsing error.
         match lockfile_error {
-            Some(err) => return Err(err),
-            None => {
-                return Err(anyhow!(
-                    "Parsing {path:?} requires lockfile generation, but it was disabled through \
-                     the CLI"
-                ))
-            },
+            Some(err) => return Err(err.into()),
+            None => return Err(ParseError::ManifestWithoutGeneration(path)),
         }
```

Comment:
nit, but I think this could be:
```rust
return Err(lockfile_error.map_or(ParseError::ManifestWithoutGeneration(path), Into::into));
```

Reply:
You can't call `Into::into` on the result itself, it needs to be called on the error.

Reply:
Not sure what you mean. `lockfile_error` isn't a `Result`, it is an instance of `anyhow::Error`, which implements `Into<ParseError>` via your `thiserror` implementation.